### PR TITLE
feat(server): debug poplulate add slots option

### PR DESCRIPTION
**Pull request by adiholden**
1. add slots <start> <end> option to debug populate command
2. fix some error prints in cluster mode

### CODE REVIEW

File: src/server/debugcmd.h [hunk 1]
```diff
@@ -12,15 +13,28 @@ class EngineShardSet;
...
+ private:
+  struct PopulateOptions {
+    uint64_t total_count = 0;
+    std::string_view prefix{"key"};
+    uint32_t val_size = 0;
+    bool populate_random_values = false;
+    struct SlotRange {
+      SlotId start = 0;
+      SlotId end = 0;
+    };
```
Comment (chakaz) [inline src/server/debugcmd.h:25]: You could use the existing `SlotRange` from `cluster_config.h`: https://shaharmike.com/dfly/dragonfly/src/server/cluster/cluster_config.h.html#dfly::ClusterConfig::SlotRange

File: src/server/debugcmd.cc [hunk 2]
```diff
@@ -306,29 +350,55 @@ void DebugCmd::Populate(CmdArgList args) {
...
+  uint64_t index = from;
+  uint64_t added = 0;
+  while (added < num_of_keys) {
+    if ((index > from) && (index % num_of_keys == 0)) {
+      index = index - num_of_keys + options.total_count;
+    }
+    StrAppend(&key, index);
+
+    if (options.slot_range.has_value()) {
+      // Each fiber will add num_of_keys. Key are in the form of <key_prefix>:<index>
```
Comment (chakaz) [inline src/server/debugcmd.cc:374]: Key**s** (plural)

File: src/server/debugcmd.cc [hunk 3]
```diff
@@ -306,29 +350,55 @@ void DebugCmd::Populate(CmdArgList args) {
...
+      // We need to make sure that different fibers will not add the same key.
+      // Fiber starting <key_prefix>:<from> to <key_prefix>:<num_of_keys-1>
+      // than continue to <key_prefix>:<total_count> to <key_prefix>:<total_count+num_of_keys-1>
```
Comment (chakaz) [inline src/server/debugcmd.cc:377]: then

File: src/server/debugcmd.cc [hunk 4]
```diff
@@ -306,29 +350,55 @@ void DebugCmd::Populate(CmdArgList args) {
...
+      // and continue untill num_of_keys are added.
```
Comment (chakaz) [inline src/server/debugcmd.cc:378]: until

File: src/server/debugcmd.cc [hunk 5]
```diff
@@ -306,29 +350,55 @@ void DebugCmd::Populate(CmdArgList args) {
...
+
+      // Add keys only in slot range.
+      SlotId sid = ClusterConfig::KeySlot(key);
+      if (sid < options.slot_range->start || sid > options.slot_range->end) {
+        key.resize(prefsize);  // shrink back
```
Comment (chakaz) [inline src/server/debugcmd.cc:383]: I'd move this `resize()` (and the one below) to be right before `StrAppend(&key, index)` above, it feels less error prone (and repetitive).

File: src/server/debugcmd.cc [hunk 6]
```diff
@@ -249,55 +250,98 @@ void DebugCmd::Load(string_view filename) {
...
   }
 
-  bool populate_random_values = false;
-  if (args.size() > 4) {
-    ToUpper(&args[4]);
-    std::string_view str = ArgS(args, 4);
-    if (str != "RAND") {
-      return (*cntx_)->SendError(kSyntaxErr);
+  size_t index = 4;
+  while (args.size() > index) {
```
Comment (chakaz) [inline src/server/debugcmd.cc:277]: Any reason not to use a `for` loop here?

**Review by chakaz — APPROVED** [review]

File: src/server/debugcmd.cc (hunk 6 above)
Comment (adiholden) [reply src/server/debugcmd.cc:277]: fixing

File: src/server/cluster_family_test.cc [hunk 7]
```diff
@@ -463,6 +463,43 @@ TEST_F(ClusterFamilyTest, ClusterGetSlotInfo) {
...
+  string config = absl::Substitute(config_template, RunAdmin({"dflycluster", "myid"}).GetString());
+
+  EXPECT_EQ(RunAdmin({"dflycluster", "config", config}), "OK");
+
+  Run({"debug", "populate", "10000", "key", "4", "SLOTS", "0", "1000"});
+
+  for (int i = 0; i <= 16383; ++i) {
+    string slot_str = absl::StrCat(i);
+    auto slots_info = RunAdmin({"dflycluster", "getslotinfo", "slots", slot_str}).GetVec();
+    EXPECT_EQ(slots_info.size(), 3);
```
Comment (chakaz) [inline src/server/cluster_family_test.cc:493]: Nit: this is unneeded given the below `ElementsAre()`

**Review by chakaz — DISMISSED** [review]

**Review by chakaz — APPROVED** [review]

**Pull request merged**